flexmeasures/data/schemas/scheduling/__init__.py

```diff
@@ -310,17 +312,19 @@ def _validate_power_fields(self, data: dict):
 
     def _validate_field(self, data: dict, field_type: str, field: str, unit_validator):
         """Validate fields based on type and unit validator."""
+        unit_type = field_type
+        field_type = "capacity price" if field_type == "capacity-price" else field_type
 
         if isinstance(data[field], ur.Quantity):
             if not unit_validator(str(data[field].units)):
                 raise ValidationError(
-                    f"{field_type.capitalize()} field '{self.mapped_schema_keys[field]}' must be a {field_type} unit.",
+                    f"{field_type.capitalize()} field '{self.mapped_schema_keys[field]}' must be a {unit_type} unit.",
```

Comment:
I changed *be* into *have*, as in: the field ('s value) *is* a quantity that *has* a unit.

```suggestion
                    f"{field_type.capitalize()} field '{self.mapped_schema_keys[field]}' must have a {unit_type} unit.",
```